From b033ef3b4b2384df4165ae7829155cafd4ee3488 Mon Sep 17 00:00:00 2001
From: Sven Anderson <sven@anderson.de>
Date: Mon, 13 Jan 2025 19:55:15 +0100
Subject: [PATCH 23/26] xhci: workaround z790 non-root-hub speed detection

---
 grub-core/bus/usb/xhci.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/grub-core/bus/usb/xhci.c b/grub-core/bus/usb/xhci.c
index dc89b9619..d13a7c39d 100644
--- a/grub-core/bus/usb/xhci.c
+++ b/grub-core/bus/usb/xhci.c
@@ -2285,6 +2285,11 @@ grub_xhci_detect_dev (grub_usb_controller_t dev, int port, int *changed)
   if (!(portsc & GRUB_XHCI_PORTSC_CCS))
     return GRUB_USB_SPEED_NONE;
 
+  if (port == 4 || port == 12 && speed < XHCI_USB_HIGHSPEED) { // inital hub speed detection on Z790 is too low
+    grub_dprintf("xhci", "%s: setting internal hub speed to high\n", __func__);
+    return GRUB_USB_SPEED_HIGH;
+  }
+
   for (grub_uint8_t i = 0; i < 16 && x->psids[port].psids[i].id > 0; i++)
     {
       if (x->psids[port].psids[i].id == speed)
-- 
2.39.5

