--- config.h	2020-12-14 01:27:24.300252326 +0000
+++ config.h.new	2020-12-14 01:27:30.004252080 +0000
@@ -17,7 +17,7 @@
 #define CONSERVATIVE_SMP 1
 
 /* BEEP_MODE - Beep on error. Default off, Change to 1 to enable */
-#define BEEP_MODE 0
+#define BEEP_MODE 1
 
 /* BEEP_END_NO_ERROR - Beep at end of each pass without error. Default off, Change to 1 to enable */
 #define BEEP_END_NO_ERROR 0
@@ -35,7 +35,7 @@
 #define SERIAL_TTY 0
 
 /* SERIAL_BAUD_RATE - Baud rate for the serial console */
-#define SERIAL_BAUD_RATE 9600
+#define SERIAL_BAUD_RATE 115200
 
 /* SCRN_DEBUG - extra check for SCREEN_BUFFER
  */ 
