From 4f52c0880bd8fbf53a49e6a29318033fbf756ee0 Mon Sep 17 00:00:00 2001
From: Leah Rowe <leah@libreboot.org>
Date: Sun, 24 Jul 2022 14:46:54 +0100
Subject: [PATCH 2/6] Lenovo X230 patch: gfx_uma_size 224M by default

---
 src/mainboard/lenovo/x230/cmos.default | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/mainboard/lenovo/x230/cmos.default b/src/mainboard/lenovo/x230/cmos.default
index 7314066c2b..2e315d4521 100644
--- a/src/mainboard/lenovo/x230/cmos.default
+++ b/src/mainboard/lenovo/x230/cmos.default
@@ -16,3 +16,4 @@ backlight=Both
 usb_always_on=Disable
 f1_to_f12_as_primary=Enable
 me_state=Normal
+gfx_uma_size=224M
-- 
2.25.1

