From 49cb9863828b90d1dcbd107f4ce4987d0567b837 Mon Sep 17 00:00:00 2001
From: Leah Rowe <leah@libreboot.org>
Date: Sun, 13 Mar 2022 18:08:57 +0000
Subject: [PATCH 3/3] specifically use python3 in gentpl.py

---
 gentpl.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gentpl.py b/gentpl.py
index c86550d4f..cb89cef37 100644
--- a/gentpl.py
+++ b/gentpl.py
@@ -1,4 +1,4 @@
-#! /usr/bin/python
+#! /usr/bin/python3
 #  GRUB  --  GRand Unified Bootloader
 #  Copyright (C) 2010,2011,2012,2013  Free Software Foundation, Inc.
 #
-- 
2.25.1

