From b8bb450bef9f9a486917115bfe78519838558300 Mon Sep 17 00:00:00 2001
From: Alexei Sorokin <sor.alexei@meowr.ru>
Date: Sun, 27 Nov 2022 18:36:26 +0300
Subject: [PATCH 15/18] lenovo/x230: fix the data.vbt path for the EDP variant

---
 src/mainboard/lenovo/x230/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/mainboard/lenovo/x230/Kconfig b/src/mainboard/lenovo/x230/Kconfig
index 4d8325ea43..409892f3ab 100644
--- a/src/mainboard/lenovo/x230/Kconfig
+++ b/src/mainboard/lenovo/x230/Kconfig
@@ -63,7 +63,7 @@ config OVERRIDE_DEVICETREE
 	default "variants/\$(CONFIG_VARIANT_DIR)/overridetree.cb"
 
 config INTEL_GMA_VBT_FILE
-	default "variants/x230_edp/data.vbt" if BOARD_LENOVO_X230_EDP
+       default "src/mainboard/\$(MAINBOARDDIR)/variants/x230_edp/data.vbt" if BOARD_LENOVO_X230_EDP
 
 config USBDEBUG_HCD_INDEX
 	int
-- 
2.39.2

